### PR TITLE
test: list widget bug and other regression bugs automation

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
Automated #19506 and #19654
ListWidget re-size bug was a bug noticed as observation which is automated.

### CODE REVIEW

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/ExplorerTests/Pages_spec.js [hunk 1]
```diff
@@ -29,10 +30,18 @@ describe("Pages", function () {
     });
   });
 
-  it("3. Checks if 404 is showing correct route", () => {
+  it("3. Check for Refrsh page and validate", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/ExplorerTests/Pages_spec.js:33]: Combine such small `it's` into another case with comment.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+describe("Dynamic Height Width validation", function() {
+  afterEach(() => {
+    agHelper.SaveLocalStorageCache();
+  });
+
+  beforeEach(() => {
+    agHelper.RestoreLocalStorageCache();
+  });
+  it("Validate change with auto height width for List widgets", function() {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js:14]: Add the bug id to `it/desc` descriptions to know the bugs are scripted.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+const dsl = require("../../../../fixtures/ResizeListDsl.json");
+const commonlocators = require("../../../../locators/commonlocators.json");
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+const agHelper = ObjectsRegistry.AggregateHelper;
+
+describe("Dynamic Height Width validation", function() {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js:6]: There are multiple failures highlighted by GitHub actions.. Check & fix all of it.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+const dsl = require("../../../../fixtures/ResizeListDsl.json");
+const commonlocators = require("../../../../locators/commonlocators.json");
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+const agHelper = ObjectsRegistry.AggregateHelper;
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js:4]: Remove unused imports.

**Review by Aishwarya-U-R — COMMENTED** [review]
Include the list of all bug id's automated in this PR in the description & its better to fill up other sections as appropriate.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js [hunk 5]
```diff
@@ -16,7 +19,18 @@ describe("Create workspace and a new app / delete and recreate app", function ()
         newWorkspaceName = interception.response.body.data.name;
         cy.renameWorkspace(newWorkspaceName, workspaceId);
       });
+      //Automated as part of Bug19506
+      if (CURRENT_REPO === REPO.CE) {
+        cy.get(application.shareButton).first().click({ force: true });
+        cy.xpath(application.placeholderTxt).should("be.visible");
+      }
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js:26]: Why have we not added validation for EE also - any reason?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+describe("Dynamic Height Width validation", function () {
+  afterEach(() => {
+    agHelper.SaveLocalStorageCache();
+  });
+
+  beforeEach(() => {
+    agHelper.RestoreLocalStorageCache();
+  });
+  it("Validate change with auto height width for List widgets", function () {
```
[the remark this answers is not in the document]
Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js:5]: Add the Id of the bug automated here.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js (hunk 5 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js:26]: If CE,EE are same - this if can be removed.

Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js:26]: There are 3 places of this share. Are all places covered?

Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js:26]: yes its taken care

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+const agHelper = ObjectsRegistry.AggregateHelper;
+
+describe("Dynamic Height Width validation", function () {
+  afterEach(() => {
+    agHelper.SaveLocalStorageCache();
+  });
+
+  beforeEach(() => {
+    agHelper.RestoreLocalStorageCache();
+  });
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js]: done

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js:5]: done

File: app/client/cypress/locators/Applications.json [hunk 8]
```diff
@@ -1,4 +1,6 @@
 {
     "copyUrl": ".t--copy-url",
-    "forkButton": ".t--fork-app"
+    "forkButton": ".t--fork-app",
+    "shareButton": "button:contains('Share')",
+    "placeholderTxt": "//input[@placeholder='Enter email address(es)']"
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/locators/Applications.json:5]: as discussed behavior is same both CE/EE

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js [hunk 9]
```diff
@@ -16,7 +21,14 @@ describe("Create workspace and a new app / delete and recreate app", function ()
...
         cy.renameWorkspace(newWorkspaceName, workspaceId);
       });
+      //Automated as part of Bug19506
+      cy.get(application.shareButton).first().click({ force: true });
+      cy.xpath(application.placeholderTxt).should("be.visible");
+      cy.reload();
       cy.CreateAppForWorkspace(workspaceId, appid);
+      cy.get(homePage.shareApp).click({ force: true });
+      cy.xpath(application.placeholderTxt).should("be.visible");
+      cy.reload();
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js]: done taken care in CreateSameAppInDiffWorkspace_spec

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js [hunk 10]
```diff
@@ -1,4 +1,9 @@
 /// <reference types="Cypress" />
+import homePage from "../../../../locators/HomePage";
+const publish = require("../../../../locators/publishWidgetspage.json");
+const application = require("../../../../locators/Applications.json");
+import { ObjectsRegistry } from "../../../../support/Objects/Registry";
+let HomePage = ObjectsRegistry.HomePage;
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js]: done

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js (hunk 3 above)
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js]: done

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js (hunk 2 above)
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/DynamicHeight/DynamicHeight_List_Resizing_spec.js]: updated the comment , there is no bug id for this but Kamakshi showed me this use-case and requested to add

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/ExplorerTests/Pages_spec.js (hunk 1 above)
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/ExplorerTests/Pages_spec.js]: done

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/DeleteWorkspace_spec.js [hunk 11]
```diff
@@ -1,6 +1,8 @@
 /// <reference types="Cypress" />
 import homePage from "../../../../locators/HomePage";
+import { REPO, CURRENT_REPO } from "../../../../fixtures/REPO";
 import * as _ from "../../../../support/Objects/ObjectsCore";
+const application = require("../../../../locators/Applications.json");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/DeleteWorkspace_spec.js:5]: Is the import of REPO, application needed in this spec?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js [hunk 12]
```diff
@@ -16,7 +19,16 @@ describe("Create workspace and a new app / delete and recreate app", function ()
         newWorkspaceName = interception.response.body.data.name;
         cy.renameWorkspace(newWorkspaceName, workspaceId);
       });
+      //Automated as part of Bug19506
+      cy.get(application.shareButton).first().click({ force: true });
+      cy.xpath(application.placeholderTxt).should("be.visible");
+      cy.reload();
       cy.CreateAppForWorkspace(workspaceId, appid);
+      if (CURRENT_REPO === REPO.CE) {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js:27]: Still dont see the `if` condition removal, any reason?

Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js]: that was oversight, updated now

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/DeleteWorkspace_spec.js (hunk 11 above)
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/DeleteWorkspace_spec.js:5]: removed

Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/DeleteWorkspace_spec.js:5]: removed

**Review by Aishwarya-U-R — APPROVED** [review]
From next PR, I would request it be written in TS format to be merged.

**Comment by Aishwarya-U-R** [comment]
@NandanAnantharamu - always merge latest & then trigger new ok-to-test. Cancelling above run.

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js [hunk 13]
```diff
@@ -16,7 +19,22 @@ describe("Create workspace and a new app / delete and recreate app", function ()
...
         cy.renameWorkspace(newWorkspaceName, workspaceId);
       });
+      //Automated as part of Bug19506
+      cy.get(application.shareButton).first().click({ force: true });
+      if (CURRENT_REPO === REPO.CE) {
+        cy.xpath(application.placeholderTxt).should("be.visible");
+      } else {
+        cy.xpath(application.placeholderTxtEE).should("be.visible");
+      }
+      cy.reload();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js:29]: Why is this reload needed?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js [hunk 14]
```diff
@@ -16,7 +19,22 @@ describe("Create workspace and a new app / delete and recreate app", function ()
...
       cy.CreateAppForWorkspace(workspaceId, appid);
+      cy.get(homePage.shareApp).click({ force: true });
+      if (CURRENT_REPO === REPO.CE) {
+        cy.xpath(application.placeholderTxt).should("be.visible");
+      } else {
+        cy.xpath(application.placeholderTxtEE).should("be.visible");
+      }
+      cy.reload();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js:37]: Why is this reload needed?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateSameAppInDiffWorkspace_spec.js [hunk 15]
```diff
@@ -29,6 +32,22 @@ describe("Create app same name in different workspace", function () {
...
+      .find(homePage.optionsIcon)
+      .click({ force: true });
+    cy.xpath(homePage.members).click({ force: true });
+    cy.get(homePage.inviteUserMembersPage).click({ force: true });
+    if (CURRENT_REPO === REPO.CE) {
+      cy.xpath(application.placeholderTxt).should("be.visible");
+    } else {
+      cy.xpath(application.placeholderTxtEE).should("be.visible");
+    }
+    cy.reload();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateSameAppInDiffWorkspace_spec.js:49]: Why is this reload needed?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/DeleteWorkspace_spec.js (hunk 11 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/DeleteWorkspace_spec.js:5]: I'm still seeing the un-required imports in this spec?

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateSameAppInDiffWorkspace_spec.js (hunk 15 above)
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateSameAppInDiffWorkspace_spec.js]: removed adn udpated

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js (hunk 13 above)
Comment (NandanAnantharamu) [reply app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Workspace/CreateAppWithSameNameInWorkspace_spec.js]: updated

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**